docs/source/extensions.rst

```diff
@@ -3,17 +3,233 @@
 Environment extensions
 ======================
 
-The environment extension is our fuctionality that allows agent to infer latent observations that are
+The environment extension is our functionality that allows agent to infer latent observations that are
 not originally supported by the environment. You can either choose one of our built-in extensions or
-implemet your own with the help of this short guide.
+implement your own with the help of this short guide.
+
+.. _key_concepts:
+
+Key concepts
+------------
+
+The main axis of this module is the :ref:`abstract class <base_ext>` ``BaseExt``, which provides the core
+functionality of extensions. It implements important methods, such as:
+
+* ``get_agent_params`` - this method is responsible for providing default arguments specified by extension
```

Comment:
Przepraszam, ale to wciąż jest trochę enigmatycznie napisane. Ja dalej nie do końca wiem co ta metoda robi, więc wydaje mi się że użytkownik będzie miał tym większe trudności ze zrozumieniem. Też ten fragment jest na początku co może go zniechęcic na dzień dobry. Jakby Ci się udało to jakoś jeszcze to sparafrazować?